src/rdf.rs

```diff
@@ -347,6 +353,53 @@ impl TryFrom<IRIOrBlankNodeIdentifier> for Predicate {
     }
 }
 
+impl Object {
+    /// Get string value, discarding type information
+    pub fn to_string_lossy(&self) -> &str {
```

Comment:
I've removed this in 9caedd2f2f5bd64ddf4ec0c03d8d29d42c83b4e5 and 11b28e2345cf29d512fead14f42c70291a1bddb8. This was a weird serialization as it discards string type information. It is because the `domain` and `challenge` proof properties don't have types defined, but could reasonably be strings or IRIs.  Other properties have more or less specific types (e.g. `verificationMethod` is a IRI, `created` is a dateTime). I've added processing of these properties with their specific types, while allowing domain and challenge to be IRI or String.